app/src/main/java/com/concordium/wallet/data/preferences/Preferences.kt

```diff
@@ -15,9 +19,145 @@ open class Preferences {
     interface Listener {
         fun onChange()
     }
+    companion object{
+        const val SHARED_PREFERENCES_ARE_ENCRYPTED ="encrypted_shared_preference"
+    }
 
     constructor(context: Context, preferenceName: String, preferenceMode: Int) {
-        sharedPreferences = context.getSharedPreferences(preferenceName, preferenceMode)
+        val authPreferences = initializeEncryptedSharedPreferences(context, SharedPreferencesKeys.PREF_FILE_AUTH)
+        if(authPreferences.getBoolean(SHARED_PREFERENCES_ARE_ENCRYPTED, false)){
```

Comment:
Android studio suggests lifting the assignment out of the `if`. I think that's a good suggestion.

Reply:
Ok